Makefile

```diff
@@ -96,8 +104,12 @@ test: deps lint test-units test-features
 
 test-all: test test-external
 
-lint: deps-goimports
-	git ls-files | grep -v '^vendor/' | grep '.go$$' | xargs goimports -l -w
+# https://golangci-lint.run/usage/install/#local-installation
+lint: deps-golangci-lint
+	golangci-lint run --new-from-rev=origin/main
```

Comment:
This seems to only review changes in the latest commit, or any local, uncommitted, changes.

Does this run the risk of hiding a bad lint change if it it manages to get checked in? We'd have to know to run `make lint-all` occasionally to make sure we're not missing something. IMO, why not lint everything all the time?

Reply:
Uhm, I guess you are referring to lint the whole checked out repo, right? (not past git history)

If so I agree, we should check all code present on each snapshot of the codebase.

Reply:
Thanks for your comments!

Yes, that one tests your changes, any new commits over usptream/main and unchecked code. 

note that this is meant only for development, it's an optimization so when you are writing code you iterate quicker. The Github action that gets run, does lint the whole repository https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/runs/3395795620?check_suite_focus=true

I can remove it if it adds confusion. I could also alternatively configure it the same way `--fast` vscode integration is recommended to be https://golangci-lint.run/usage/integrations/ 

What do you think?

Reply:
My "strong opinion, loosely held" is that we should make sure that we can't hide bad commits. If the CI lint step runs over the whole codebase, that should be fine enough for me.

That being said, my preference is to run everything always (granted that it's fast enough) and rely (require?) on the IDE integrations to run the checks automatically. I'm setting up my GoLand to do that now.

Reply:
> My "strong opinion, loosely held" is that we should make sure that we can't hide bad commits. If the CI lint step runs over the whole codebase, that should be fine enough for me.

In development, in your branch, I can't see how it will hide bad commits.

> That being said, my preference is to run everything always (granted that it's fast enough) and rely (require?)

it's fairly fast since it has a local cache and so on. note that this makefile target has nothing to do with [editor integrations](https://golangci-lint.run/usage/integrations/), it's just a convenience make target in case you don't/can't run them in the editor.

In any case, I do not have strong opinion here and in general I don't like early optimizations, even if I do them :), so I'll make `make lint` run all of them and remove lint-all.

Thanks!